src/probe/guided_search_iterator.jl

```diff
@@ -0,0 +1,68 @@
+
+@programiterator GuidedSearchIterator(
+    spec::Vector{<:IOExample},
+    symboltable::SymbolTable,
+)
+Base.@kwdef mutable struct GuidedSearchState
```

Comment:
Could you possibly add some documentation? Describe it like I am 5, and put it above the definition. Preferably with a reference to the paper.

Reply:
I added some documentation now. It is good?